spec.md

```diff
@@ -132,18 +132,18 @@ A web-friendly display name is camel-cased with spaces and punctuation supported
 
 | Response field | Type | Description |
 | --- | --- | --- |
-| id | string | The id of the Oauth client that the dashboard will use. |
-| secret | string | A secret for the dashboard client |
+| id | string | The id of the Oauth client that the dashboard will use. If present, MUST be a non-empty string. |
+| secret | string | A secret for the dashboard client. If present, MUST be a non-empty string. |
 | redirect_uri | string | A URI for the service dashboard. Validated by the OAuth token server when the dashboard requests a token. |
 
 
 ##### Plan Object
 
 | Response field | Type | Description |
 | --- | --- | --- |
-| id* | string | An identifier used to correlate this plan in future requests to the broker. This MUST be globally unique within a platform marketplace. Using a GUID is RECOMMENDED. |
-| name* | string | The CLI-friendly name of the plan. MUST be unique within the service. All lowercase, no spaces. |
-| description* | string | A short description of the plan. |
+| id* | string | An identifier used to correlate this plan in future requests to the broker. This MUST be globally unique within a platform marketplace. MUST be a non-empty string. Using a GUID is RECOMMENDED. |
+| name* | string | The CLI-friendly name of the plan. MUST be unique within the service. MUST be a non-emptry string containing only lowercase alphanumeric characters, no spaces. |
```

Comment:
empty is spelled incorrectly

I would suggest `All lowercase, no spaces. MUST be a non-empty string.`

The reason is that that the snakecase guidelines for name is meant to provide a better CLI UX, but CF doesn't currently validate it other than it is a string. Hyphens and underscores should be ok. I wouldn't want to say that this field must only be alphanumeric. However, if we're going to open that can of worms, it should be addressed for a number of other display name fields (e.g. service name). Instead, I'd recommend leaving the guidance as "All lowercase, no spaces."